File: housing/src/interfaces/tenant.ts
import { Car } from './car';
import { Batch } from './batch';
import { TenantAddress } from './tenant-address';

export interface Tenant {
  id: string;
  email: string;
  gender: string;
  firstName: string;
  lastName: string;
  addressId: string;
  roomId: string;
  carId: string;
  batchId: number;
  tenantAddress: TenantAddress;
  car: Car;
  batch: Batch;
  trainingCenter: string;
}

File: tenant/src/Revature.Tenant.DataAccess/Entities/Tenant.cs
using System;

namespace Revature.Tenant.DataAccess.Entities
{
  /// <summary>
  /// This class defines a data access entity tenant.
  /// This is the tenant object we access through our database.
  /// </summary>
  public class Tenant
  {
    public Guid Id { get; set; }
    public string Email { get; set; }
    public string Gender { get; set; }
    public string FirstName { get; set; }
    public string LastName { get; set; }
    public Guid AddressId { get; set; }
    public Guid RoomId { get; set; }
    public int? CarId { get; set; }
    public virtual Car Car { get; set; }
    public int? BatchId { get; set; }
    public virtual Batch Batch { get; set; }
    public Guid TrainingCenter { get; set; }
  }
}
